### ISSUE TITLE
iPAS AI 應用規劃師科目3練習 - 2026-08-31

### ISSUE THREAD

**Issue by JohnnyLai92** [JohnnyLai92/ipas#16]
# iPAS AI 應用規劃師科目3練習 - 2026-08-31（第 6 輪）

## 練習說明

本次共 **10 題**，涵蓋科目3「機器學習技術與應用」，本輪主要側重 **L232 機器學習與深度學習**，次要為 **L233 建模與參數調校**，並以其他領域補齊跨域涵蓋。情境題以「醫療與健康照護」與「製造與工業檢測」兩種產業為背景。

- 題型：**單選 6 題、多選 2 題、情境分析 2 題**。
- 作答方式：**直接點選下方每個選項前的核取方塊**即可作答，**不需留言、不需進入編輯畫面**。單選題請只勾選一項；多選題請依標題指示勾選指定項數。
- 每題僅標示題型與考點 L 代碼，**未提供正確答案**，請自行判斷。
- 全部作答完成後，請將本 issue 的 **label 改為 `ai-analysis`**，系統即會進行批改與錯題分析。

## 第 1 題〔單選〕L23201 機器學習原理

【製造與工業檢測】某面板廠累積了 500 張已標記的瑕疵影像，另有 50,000 張尚未標記的產線影像。團隊希望**同時**運用這少量已標記與大量未標記資料來提升模型泛化能力。此情境最符合下列哪一種學習範式？

- [x] (A) 半監督式學習（Semi-Supervised Learning）：結合少量標記與大量未標記資料
- [ ] (B) 純監督式學習：只使用 500 張已標記影像，其餘捨棄
- [ ] (C) 非監督式學習：完全忽略標記，只做分群
- [ ] (D) 強化學習：以環境獎勵訊號逐步學習策略

---

## 第 2 題〔單選〕L23202 常見機器學習演算法

【醫療與健康照護】某醫院要建立輔助診斷模型，院方要求模型的判斷能以「明確的 if-then 決策規則」呈現，方便醫師理解並供稽核。在下列演算法中，**可解釋性與決策規則呈現**最符合此需求的是？

- [ ] (A) 支援向量機（RBF 核）
- [ ] (B) 深度神經網路
- [x] (C) 決策樹（Decision Tree）
- [ ] (D) K-Means

---

## 第 3 題〔單選．程式判讀〕L23203 深度學習原理與框架

閱讀下列 PyTorch 程式片段，判斷輸出的張量形狀：

```python
import torch
import torch.nn as nn

model = nn.Linear(8, 3)
x = torch.randn(16, 8)
y = model(x)
print(y.shape)
```

- [ ] (A) torch.Size([16, 8])
- [ ] (B) torch.Size([16, 3])
- [ ] (C) torch.Size([3, 16])
- [x] (D) torch.Size([8, 3])

---

## 第 4 題〔單選〕L23301 數據準備與特徵工程

【製造與工業檢測】資料集中有一欄「產線機台代號」，其值為 M1、M2、M3、M4，彼此**沒有大小或順序關係**。若後續要餵入對數值大小敏感的模型（如 KNN、邏輯回歸），下列何種編碼方式最恰當？

- [x] (A) Label Encoding，將 M1/M2/M3/M4 轉為 0/1/2/3
- [ ] (B) 對該欄直接做標準化（Standardization）
- [ ] (C) 直接刪除該欄，因類別欄無法使用
- [ ] (D) One-Hot Encoding，將每個機台代號轉為獨立的 0/1 欄位

---

## 第 5 題〔單選〕L23303 模型評估指標

【製造與工業檢測】某設備故障預警模型部署於產線安全監控，經營團隊強調「**絕對不能漏掉任何一台真正即將故障的設備**」，寧可多發幾次誤報也要把真正故障全數抓出。此需求應優先改善下列哪一個評估指標？

- [ ] (A) Precision（精確率）
- [ ] (B) Accuracy（整體正確率）
- [x] (C) Recall（召回率）
- [ ] (D) 讓 F1-score 固定維持在 0.5

---

## 第 6 題〔單選〕L23103 數值優化技術與方法

某深度學習模型在訓練時，Loss 曲線劇烈震盪、不但沒有下降，甚至持續變大而發散。在其他設定正常的前提下，最可能的原因與對策為何？

- [ ] (A) 學習率（Learning Rate）過低，應大幅調高
- [x] (B) 學習率（Learning Rate）過高，應調小
- [ ] (C) 模型已經收斂，此為正常現象
- [ ] (D) 資料量太少，必須立刻增加網路層數

---

## 第 7 題〔多選（可勾選 4 項）〕L23304 模型調整與優化

【製造與工業檢測】某瑕疵分類模型 Training Accuracy 高達 99%，但 Validation Accuracy 僅約 70%，且差距持續擴大。針對此現象，下列哪些是**合理有效**的改善做法？（請勾選 4 項）

- [x] (A) 加入 L2 正則化（Regularization）
- [x] (B) 加入 Dropout
- [x] (C) 使用 Early Stopping
- [x] (D) 增加訓練資料或使用資料擴增（Data Augmentation）
- [ ] (E) 大幅增加網路層數與參數量，讓模型更複雜
- [ ] (F) 移除所有正則化並無限延長訓練

---

## 第 8 題〔多選（可勾選 3 項）〕L23203 深度學習原理與框架

下列關於深度學習模型與概念的敘述，哪些**正確**？（請勾選 3 項）

- [x] (A) CNN 常用於影像辨識，透過卷積運算擷取空間特徵
- [x] (B) LSTM 相較於一般 RNN，較能處理長期序列依賴問題
- [ ] (C) Transfer Learning 可在小型資料集上運用預訓練模型，降低訓練成本
- [x] (D) ReLU 會把所有輸入（含負值與正值）都壓縮到 0～1 之間
- [ ] (E) 只要增加 Batch Size，模型準確率就一定會提升

---

## 第 9 題〔情境分析．單選〕L23401 數據隱私、安全與合規

【醫療與健康照護】某醫學中心計畫以病患的歷史病歷、檢驗數據與就診紀錄，訓練一個「30 天再住院風險預測模型」。專案啟動時，資料工程團隊拿到了含姓名、身分證號與完整病歷的原始資料表。在正式投入建模之前，團隊**最優先且最重要**應該做的是？

- [x] (A) 先確認資料使用的合法性與知情同意，落實去識別化、資料最小化與存取權限控管
- [ ] (B) 為加速開發，直接把含姓名的完整病歷開放給全院人員取用
- [ ] (C) 只要模型準確率夠高就好，隱私與合規不是這階段的重點
- [ ] (D) 為節省時間，略過知情同意與去識別化，直接開始訓練

---

## 第 10 題〔情境分析．單選〕L23202 常見機器學習演算法

【製造與工業檢測】某工廠只有約 2,000 張已標記的瑕疵產品影像，且 GPU 運算資源有限，但希望能**快速**建立一個成效不錯的影像分類模型。在下列策略中，最適合此情境（小資料集、資源有限、需快速上線）的是？

- [ ] (A) 從零開始訓練一個超大型深度神經網路
- [x] (B) 改用 K-Means 非監督分群取代影像分類任務
- [ ] (C) 放棄影像模型，全部改回人工全檢
- [ ] (D) 採用 Transfer Learning，微調（Fine-tune）預訓練好的 CNN 模型

---

作答完成後，記得將 label 改為 `ai-analysis` 進行批改。

**Comment by JohnnyLai92** [comment]
**Claude Code 練習批改與題型分析**

## 批改結果

| 題號 | 你的作答 | 正確答案 | 判定 |
|---|---|---|---|
| 1 | A | **A** | ✅ 正確 |
| 2 | C | **C** | ✅ 正確 |
| 3 | D | **B** | ❌ 錯誤 |
| 4 | A | **D** | ❌ 錯誤 |
| 5 | C | **C** | ✅ 正確 |
| 6 | B | **B** | ✅ 正確 |
| 7 | A,B,C,D | **A,B,C,D** | ✅ 正確 |
| 8 | A,B,D | **A,B,C** | ❌ 錯誤 |
| 9 | A | **A** | ✅ 正確 |
| 10 | B | **D** | ❌ 錯誤 |

**總分：6 / 10**

### 逐題解析

- **第 1 題 ✅**：少量標記 + 大量未標記資料同時使用，正是**半監督式學習**的定義。
- **第 2 題 ✅**：決策樹能輸出明確的 if-then 規則，可解釋性最高，適合稽核。SVM(RBF)、DNN 為黑箱；K-Means 為分群非分類。
- **第 3 題 ❌**：`nn.Linear(8, 3)` 表示「輸入 8 維 → 輸出 3 維」。輸入 `x` 形狀為 (16, 8)，batch=16 不變，特徵維度由 8 映射到 3，故輸出為 **(16, 3)，答案 B**。你選的 D `[8, 3]` 是把「權重矩陣形狀」誤當成輸出。
- **第 4 題 ❌**：機台代號 M1~M4 為**無順序的名目類別**。若用 Label Encoding 轉 0/1/2/3，會讓 KNN、邏輯回歸等對數值大小敏感的模型誤以為 M4 > M1 有大小關係。正確做法是 **One-Hot Encoding（D）**，各類別轉獨立 0/1 欄位，避免虛假的順序關係。
- **第 5 題 ✅**：「絕對不能漏掉真正故障」＝盡量降低偽陰性 → 優先提升 **Recall（召回率）**。
- **第 6 題 ✅**：Loss 震盪且發散，典型為**學習率過高**，應調小。
- **第 7 題 ✅**：A~D 皆為對抗過擬合的合理手段；E、F 反而加劇過擬合，正確排除。
- **第 8 題 ❌**：正確為 **A、B、C**。
  - (C) Transfer Learning 在小資料集上運用預訓練模型、降低成本，敘述正確，**你漏勾了**。
  - (D) ReLU 為 `max(0, x)`，只把負值截為 0、正值原樣輸出，**並非壓縮到 0~1**（那是 Sigmoid），敘述錯誤，**你多勾了**。
  - (E) 增大 Batch Size 不保證提升準確率，正確排除。
- **第 9 題 ✅**：醫療個資建模前，**最優先**應確認合法性、知情同意，落實去識別化、資料最小化與權限控管。
- **第 10 題 ❌**：小資料集（2,000 張）、資源有限、需快速上線 → **Transfer Learning 微調預訓練 CNN（D）**最適合。你選的 B（K-Means）是非監督分群，無法取代有標記的瑕疵分類任務，且效果通常不佳。

---

### 題型分析

錯的 4 題集中在三個弱點：

1. **深度學習框架與基礎觀念（L23203，第 3、8 題）**
   - 對 `nn.Linear(in, out)` 的形狀轉換規則不熟：**batch 維不變，只有特徵維被映射**。
   - 混淆激勵函數：**ReLU = max(0, x)**（非負截斷）；把值壓到 0~1 的是 **Sigmoid**。這是常見陷阱題。

2. **特徵工程的編碼選擇（L23301，第 4 題）**
   - 名目型（無序）類別 → **One-Hot**；有序類別（如小/中/大）→ Label/Ordinal Encoding。判斷關鍵在「類別是否有順序關係」與「模型是否對數值大小敏感」。

3. **演算法選型的情境判斷（L23202，第 10 題）**
   - 遇到「小資料 + 資源有限 + 快速上線 + 分類任務」，標準答案幾乎都是 **Transfer Learning + Fine-tune**。要能把任務型態（分類 vs 分群）與資料條件對應到正確策略。

---

### 新的解答建議

**修正版作答：**
- 第 3 題：改選 **(B) torch.Size([16, 3])**
- 第 4 題：改選 **(D) One-Hot Encoding**
- 第 8 題：改勾 **(A)(B)(C)**，取消 (D)
- 第 10 題：改選 **(D) Transfer Learning，微調預訓練 CNN**

**讀書建議：**
- **激勵函數對照表**：Sigmoid(0~1)、Tanh(−1~1)、ReLU(max(0,x))、Softmax(機率分布)，各自值域與用途務必記牢。
- **PyTorch 形狀練習**：實際跑 `nn.Linear`、`nn.Conv2d` 觀察 in/out 維度變化，建立「(N, features) → (N, out)」的直覺。
- **編碼決策流程**：先問「類別有無順序」→ 無序用 One-Hot、有序用 Ordinal；再考量維度爆炸時可用目標編碼等進階法。
- **選型口訣**：小資料/少資源/要快 → **遷移學習**；要可解釋 → **決策樹**；要抓全（不漏） → **Recall**；要準（不誤報） → **Precision**。

繼續加油，L232 的深度學習基礎再鞏固一輪，下次上 8 分沒問題！💪

**Comment by JohnnyLai92** [comment]
已完成錯題分析並輸出到 `docs/ipas-ai-practice-issue-16.md`。

## Claude Code 歸檔摘要
已建立 `docs/ipas-ai-practice-issue-16.md`，未修改其他檔案。

**本次成績 6／10**，依 issue 勾選結果與批改留言判定，答錯 4 題：

| 題號 | 考點 | 你的作答 | 正確答案 |
|---|---|---|---|
| 3 | L23203 深度學習框架 | D [8,3] | **B [16,3]** |
| 4 | L23301 特徵編碼 | A Label Encoding | **D One-Hot Encoding** |
| 8 | L23203 深度學習觀念 | A,B,D | **A,B,C** |
| 10 | L23202 演算法選型 | B K-Means | **D Transfer Learning** |

檔案內容涵蓋：基本資訊、錯題列表、每題的錯誤原因／正確觀念／相關機器學習考點，以及後續複習建議。三大弱點歸納為 **L23203 深度學習基礎**（`nn.Linear` 形狀規則、ReLU vs Sigmoid 激勵函數）、**L23301 名目類別編碼**（無序用 One-Hot）、**L23202 演算法選型**（小資料+快速上線→遷移學習）。

格式沿用既有 issue-14 的歸檔風格（表格 + 逐題分析 + 複習建議），並附上修正版作答。
